R/recipe_run.R

```diff
@@ -530,30 +568,43 @@ orderly_prepare_data <- function(config, info, parameters, envir) {
   } else {
     hash_global <- NULL
   }
+  ret <- list(hash_resources = hash_resources, hash_global = hash_global)
 
-  if (length(missing_packages) > 0) {
-    stop(paste("Missing packages:", 
-               paste(squote(missing_packages), collapse = ", ")))
-  }
+  ret
+}
 
-  ret <- list(data = ldata, hash_resources = hash_resources,
-              hash_global = hash_global, n_dev = n_dev)
+compare_resource_hashes <- function(pre_run_hashes, post_run_hashes) {
+  ## compare resource hashes here
+  found <- names(post_run_hashes)
+  expected <- names(pre_run_hashes)
 
-  if (!is.null(info$connection)) {
-    ret$con <- data[[info$connection]]
+  ## we expected a resource but can't find it.
+  not_found <- which(!(expected %in% found))
+  if (any(not_found)) {
+    stop("Script either did not copy or deleted resources: ",
```

Comment:
why "did not copy"? The script should not copy a resource

Reply:
That should probably read Orderly has failed to copy across a resource OR the script has deleted a resources file.
The failure to copy is caught elsewhere so I'll change this to `Script deleted resources`